src/Orchard.Web/Modules/Orchard.OpenId/Module.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Orchard.Environment.Shell;
+using Orchard.FileSystem.AppData;
+using Orchard.OpenId.Indexes;
+using Orchard.OpenId.Models;
+using Orchard.OpenId.Services;
+using Orchard.Security;
+using Orchard.Settings;
+using Orchard.Users.Models;
+using System;
+
+namespace Orchard.OpenId
+{
+    public class Startup : StartupBase
+    {     
+        private readonly string _certificateFullPath;
+        private const string certificateFileName = "Certificate.pfx";
+        private readonly string tenantUrlPrefix;
+        private readonly string tenantName;
+        public Startup(ShellSettings shellSettings, IAppDataFolder appDataFolder, ILoggerFactory loggerFactory)   
+        {
+            tenantName = shellSettings.Name;
+            tenantUrlPrefix = string.IsNullOrWhiteSpace(shellSettings.RequestUrlPrefix) ? "" : "/" + shellSettings.RequestUrlPrefix;
+            _certificateFullPath = appDataFolder.Combine(shellSettings.Name, certificateFileName);
+        }
+
+        public override void Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var baseUrl = serviceProvider.GetRequiredService<ISiteService>().GetSiteSettingsAsync().Result.BaseUrl;
+            baseUrl = baseUrl.EndsWith("/") ? baseUrl.Remove(baseUrl.Length - 1) : baseUrl;
+            var tenantUrl = baseUrl + tenantUrlPrefix;
+
+            builder.UseOpenIddict();
+            builder.UseJwtBearerAuthentication(new JwtBearerOptions()
+            {
+                AutomaticAuthenticate = true,
+                AutomaticChallenge = true,
+#if DEBUG
+                RequireHttpsMetadata = false,
+#endif
+                Audience = tenantUrl,
+                Authority = tenantUrl
+            });
+        }
+
+        public override void ConfigureServices(IServiceCollection serviceCollection)
```

Comment:
Cool then.

I don't suppose Orchard automatically overrides the data protection provider that are used by the different tenants? I took a brief look to the users module, and I didn't find any trace of a custom data protection provider that isolates the Identity instances across the tenants (which is bad for the reasons I mentioned earlier).